### PR TITLE
remember section id on course page version selector

### PR DESCRIPTION
**Pull request by davidsbailey**
previously, when switching between course versions on the course page, we'd forget the section id. this PR preserves the section id, and adds some ui tests to confirm this as well as confirm basic course page navigation with courseVersions experiment enabled.

### CODE REVIEW

**Comment by davidsbailey** [comment]
"after" screenshot (the url changes a few sec before the dropdown):
course-page-version

File: dashboard/test/ui/features/teacher_homepage.feature [hunk 1]
```diff
@@ -62,3 +62,28 @@ Feature: Using the teacher homepage sections feature
     And the href of selector "a:contains(View Unit Overview)" contains the section id
     And I wait until element ".header_popup_body .uitest-ProgressBubble:first" is visible
     And the href of selector ".header_popup_body .uitest-ProgressBubble:first" contains the section id
+
+  Scenario: Navigate to course pages with course versions enabled
+    Given I am on "http://studio.code.org/home?enableExperiments=courseVersions"
+    When I see the section set up box
+    # By default, the '18-'19 version should be selected.
```
Comment (caleybrock) [inline dashboard/test/ui/features/teacher_homepage.feature:69]: Will these UI tests need to be updated as we add new script versions? Anything we can do to make it more evergreen?

File: apps/src/templates/courseOverview/CourseOverview.js [hunk 2]
```diff
@@ -98,7 +101,7 @@ export default class CourseOverview extends Component {
               onChange={this.onChangeVersion}
               value={name}
               style={styles.versionDropdown}
-              className="version-selector"
+              id="version-selector"
```
Comment (caleybrock) [inline apps/src/templates/courseOverview/CourseOverview.js:104]: is this for UI tests only? I've seen us do `uitest-version-selector` to indicate its for testing so it doesn't get removed by accident.

Comment (davidsbailey) [reply apps/src/templates/courseOverview/CourseOverview.js:104]: nope, this one is for both ui tests and unit tests. 

going forward, I do not understand the rationale for this convention -- by definition a test which depends on this kind of id or classname will prevent it from being removed. by contrast a classname which only affects style/appearance is much more likely to break things and go unnoticed, unless an eyes test happens to be in place. so I'm inclined to relax this convention unless there is a reason not to. if there is a benefit i'm missing please let me know :-)

File: dashboard/test/ui/features/teacher_homepage.feature (hunk 1 above)
Comment (davidsbailey) [reply dashboard/test/ui/features/teacher_homepage.feature:69]: yes, these would need to be updated when we add new course versions. it's hard to say whether we should automate this -- it seems important to test what the user actually sees, and when we go to make version `'19-'20` visible and stable, it seems ok to have to update a few UI tests. If we do want to future-proof this, an approach for doing so isn't jumping out at me but I'm open to ideas.

**Pull request merged**